### PR TITLE
WIP :construction: (react-native-batch) create the expo plugin for Batch

### PR DESCRIPTION
**Pull request by Spoutnik97**

### CODE REVIEW

**Comment by Spoutnik97** [comment]
Hi @EvanBacon ! I saw you develop lots of plugins, so maybe you could help me to finish this one please ? 
I posted my problem : https://forums.expo.dev/t/custom-plugin-for-react-native-batch-needs-google-services-json/59627

File: packages/react-native-batch/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "prepublishOnly": "expo-module prepublishOnly",
+    "expo-module": "expo-module"
+  },
+  "keywords": [
+    "react",
+    "expo",
+    "config-plugins",
+    "prebuild",
+    "react-native-batch",
+    "expo-42"
```
Comment (EvanBacon) [inline packages/react-native-batch/package.json:27]:
```suggestion
```

File: packages/react-native-batch/src/__tests__/withReactNativeBatch-test.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { AndroidConfig, XML } from '@expo/config-plugins'
+
+import withReactNativeBatch from '../withReactNativeBatch'
+import {buildGradleFixture} from "../__tests__/fixtures/buildGradle"
+
+
+const { readAndroidManifestAsync } = AndroidConfig.Manifest
+
+
```
Comment (EvanBacon) [inline packages/react-native-batch/src/__tests__/withReactNativeBatch-test.ts:9]:
```suggestion
```

File: packages/react-native-batch/src/__tests__/withReactNativeBatch-test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('addFineControlPermissionToManifest', () => {
+  it(`adds element`, async () => {
+    const result = withReactNativeBatch(buildGradleFixture, {})
+    let gradle = ;
```
Comment (EvanBacon) [inline packages/react-native-batch/src/__tests__/withReactNativeBatch-test.ts:14]: ??

File: packages/react-native-batch/src/__tests__/withReactNativeBatchAppBuildGradle-test.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { pushDependencies } from "../withReactNativeBatchAppBuildGradle";
+import {
+  buildGradleFixture,
+  buildGradleExpectedFixture,
+} from "./fixtures/buildGradle";
+
+describe("pushDependencies", () => {
```
Comment (EvanBacon) [inline packages/react-native-batch/src/__tests__/withReactNativeBatchAppBuildGradle-test.ts:7]:
```suggestion
describe(pushDependencies, () => {
```

Comment (EvanBacon) [reply packages/react-native-batch/src/__tests__/withReactNativeBatchAppBuildGradle-test.ts:7]: do same for all tests of specific functions

File: packages/react-native-batch/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# @config-plugins/react-native-batch
```
Comment (EvanBacon) [inline packages/react-native-batch/README.md:1]: Rename the package to `@config-plugins/react-native-batch-push`

**Comment by EvanBacon** [comment]
This repo is for out-of-tree plugins with the intention of eventually getting them upstreamed. Upstreaming plugins can be tricky and much larger packages, but it looks like [`react-native-batch`](https://www.npmjs.com/package/react-native-batch) and [`react-native-batch-push`](https://www.npmjs.com/package/react-native-batch-push) both have one weekly download, it might be better to publish this plugin directly inside the upstream package/repo.

**Comment by Spoutnik97** [comment]
Thank you @EvanBacon  for your proofreading. Indeed, I had not finished my PR. 
Now it works well!
I followed your advice to create a new pull request on the Batch repository.

I have a last question : 
currently I have to create a react-ntaive.config.js file at the root of my project to add the following lines: 

```
  "react-native-batch": {
      platforms: {
        android: {
          packageInstance: "new RNBatchPackage(this.getApplication())",
        },
      },
    },
```

How do you implement this in the plugin ? Do I have to create a new mod for the MainApplication and insert `packages.add(new RNBatchPackage(this.getApplication()))`
into 
```
 @Override
    protected List<ReactPackage> getPackages() {
      List<ReactPackage> packages = new PackageList(this).getPackages();
      packages.add(new ModuleRegistryAdapter(mModuleRegistryProvider));
      return packages;
    }
```

Do you see a better way to add this package ?

File: packages/react-native-batch/src/__tests__/withReactNativeBatchAppBuildGradle-test.ts (hunk 4 above)
Comment (Spoutnik97) [reply packages/react-native-batch/src/__tests__/withReactNativeBatchAppBuildGradle-test.ts:7]: ok, I didn't know I can give a function as parameter

File: packages/react-native-batch/src/__tests__/withReactNativeBatch-test.ts (hunk 3 above)
Comment (Spoutnik97) [reply packages/react-native-batch/src/__tests__/withReactNativeBatch-test.ts:14]: Work in progress...

**Comment by EvanBacon** [comment]
@Spoutnik97 modifying the `react-native.config.js` should *not* be done inside of a config plugin. Config plugins should be used for files inside of the `ios` and `android` folders. Instead you should document that modifying the RNConfig is an extra step required for setup.

**Pull request closed without merging**